dali/test/python/test_operator_decoders_image.py

```diff
@@ -119,22 +135,57 @@ def check(img_type, size, device, threads):
         for size in {1, 10}:
             yield check, img_type, size, device, threads
 
+batch_size_test = 16
 
+@pipeline_def(batch_size=batch_size_test, device_id=0, num_threads=4)
+def img_decoder_pipe(device, out_type, files):
+    encoded, _ = fn.readers.file(files=files)
+    decoded = fn.decoders.image(encoded, device=device, output_type=out_type)
+    return decoded
 
-batch_size_alias_test=16
-
-@pipeline_def(batch_size=batch_size_alias_test, device_id=0, num_threads=4)
+def _testimpl_image_decoder_consistency_multichannel(files):
+    compare_pipelines(img_decoder_pipe("cpu", out_type=types.RGB, files=files),
+                      img_decoder_pipe("mixed", out_type=types.RGB, files=files),
+                      batch_size=batch_size_test, N_iterations=3,
+                      eps = 1e-03)
+    compare_pipelines(img_decoder_pipe("cpu", out_type=types.ANY_DATA, files=files),
+                      img_decoder_pipe("mixed", out_type=types.ANY_DATA, files=files),
+                      batch_size=batch_size_test, N_iterations=3,
+                      eps = 1e-03)
+
+def test_image_decoder_consistency_multichannel_tiff():
+    files = get_img_files(os.path.join(test_data_root, "db/single/multichannel/tiff_multichannel"), ext='tif', subdir='')
+    _testimpl_image_decoder_consistency_multichannel(files)
+
+def test_image_decoder_consistenty_multichannel_with_alpha():
+    # png and jp2 images with alpha channel
+    files = get_img_files(os.path.join(test_data_root, "db/single/multichannel/with_alpha"), subdir='')
+    _testimpl_image_decoder_consistency_multichannel(files)
+
+def _testimpl_image_decoder_consistency(img_format, img_out_type):
+    files = get_img_files(os.path.join(test_data_root, good_path, img_format))
```

Comment:
Does it make sense to limit the files to 10 or so?
Or it is better to run the test for all of them?

Reply:
I don't see the need of limiting the files we send to the reader, we'll test more or less files depending on the batch size and number of iterations